common/djangoapps/lang_pref/api.py

```diff
@@ -45,3 +46,16 @@ def released_languages():
     ]
 
     return released_languages
+
+
+def all_languages():
+    """Retrieve the list of all languages, translated and sorted.
+
+    Returns:
+        list of (language code (str), language name (str)): the language names
+        are translated in the current activated language and the results sorted
+        alphabetically.
+
+    """
+    languages = [(lang[0], _(lang[1])) for lang in settings.ALL_LANGUAGES]  # pylint: disable=translation-of-non-string
+    return sorted(languages, key=lambda l: l[1])
```

Comment:
Nit: For better code readability, consider using a variable name of length > 1.  For example, `lang`.  The current choice of `l` can be misread as the #1.  :)